### PR TITLE
[WIP] Memory profiler prototype

### PR DESCRIPTION
**Pull request by xinrong-meng**

### CODE REVIEW

**Comment by xinrong-meng** [comment]
```
  File "/Users/xinrong.meng/spark/python/lib/pyspark.zip/pyspark/worker.py", line 735, in profiling_func
    accumulator.add({result_id: (codemap_dict, None)})
  File "/Users/xinrong.meng/spark/python/lib/pyspark.zip/pyspark/accumulators.py", line 173, in add
    self._value = self.accum_param.addInPlace(self._value, term)
  File "/Users/xinrong.meng/spark/python/lib/pyspark.zip/pyspark/sql/profiler.py", line 48, in addInPlace
    PStatsParam.addInPlace(orig_perf, perf),
  File "/Users/xinrong.meng/spark/python/lib/pyspark.zip/pyspark/profiler.py", line 263, in addInPlace
    value1.add(value2)
AttributeError: 'dict' object has no attribute 'add'

```

File: python/pyspark/worker.py [hunk 1]
```diff
@@ -713,6 +713,31 @@ def profiling_func(*args, **kwargs):
...
+
+    def profiling_func(*args, **kwargs):
+        profiler = UDFLineProfiler()
+        (sub_lines, start_line) = inspect.getsourcelines(f.__code__)
+        wrapped = profiler(f, sub_lines=sub_lines, start_line=start_line)
+        ret = wrapped(*args, **kwargs)
+        codemap_dict = {
+            filename: list(line_iterator) for filename, line_iterator in profiler.code_map.items()
+        }
+        accumulator.add({result_id: (codemap_dict, None)})
```
Comment (ueshin) [inline python/pyspark/worker.py]:
```suggestion
        accumulator.add({result_id: (None, codemap_dict)})
```

Comment (xinrong-meng) [reply python/pyspark/worker.py]: Thanks!

**Comment by xinrong-meng** [comment]
```
23/12/21 13:44:12 ERROR Executor: Exception in task 5.0 in stage 0.0 (TID 5) 16]
org.apache.spark.api.python.PythonException: Traceback (most recent call last):
  File "/opt/miniconda3/envs/dev/lib/python3.9/inspect.py", line 1006, in getsourcelines
    lines, lnum = findsource(object)
  File "/opt/miniconda3/envs/dev/lib/python3.9/inspect.py", line 835, in findsource
    raise OSError('could not get source code')
OSError: could not get source code
```

**Comment by xinrong-meng** [comment]
image

**Pull request closed without merging**